independent-projects/qute/core/src/main/java/io/quarkus/qute/EngineBuilder.java

```diff
@@ -58,6 +59,11 @@ public final class EngineBuilder {
         this.timeout = 10_000;
         this.useAsyncTimeout = true;
         this.listeners = new ArrayList<>();
+        // Load EngineListener implementations via Java SPI and register them.
+        ServiceLoader<EngineListener> listenerExtensions = ServiceLoader.load(EngineListener.class);
```

Comment:
Apparently this is an inner class, making it hard to add a services file for it, because `EngineBuilder$EngineListener` is often an annoying character to deal with from the command-line.

Reply:
I know but I don't want to change the existing API.

Reply:
@mkouba I removed the service loader.